lib/core/managementClient.ts

```diff
@@ -532,6 +615,190 @@ export class ManagementClient extends LinkEntity {
+      messageBody["session-state"] = toBuffer(state);
+      const request: AmqpMessage = {
+        body: messageBody,
+        reply_to: this.replyTo,
+        application_properties: {
+          operation: Constants.operations.setSessionState
+        }
+      };
+      request.application_properties![Constants.trackingId] = generate_uuid();
+      log.mgmt("[%s] Set Session state request body: %O.", this._context.namespace.connectionId,
+        request.body);
+      log.mgmt("[%s] Acquiring lock to get the management req res link.",
+        this._context.namespace.connectionId);
+      await defaultLock.acquire(this.managementLock, () => { return this._init(); });
+      await this._mgmtReqResLink!.sendRequest(request);
+    } catch (err) {
+      const error = translate(err);
+      log.error("An error occurred while sending the renew lock request to $management " +
+        "endpoint: %O", error);
+      throw error;
+    }
+  }
+
+  /**
+   * Gets the state of the specified session.
+   * @param sessionId The session for which the state needs to be retrieved.
+   * @returns Promise<any> The state of that session
+   */
+  async getSessionState(sessionId: string): Promise<any> {
+    if (typeof sessionId !== "string") {
+      throw new Error("'sessionId' is a required parameter and must be of type 'string'.");
+    }
+    try {
+      const messageBody: any = {};
+      messageBody[Constants.sessionIdMapKey] = sessionId;
+      const request: AmqpMessage = {
+        body: messageBody,
+        reply_to: this.replyTo,
+        application_properties: {
+          operation: Constants.operations.getSessionState
+        }
+      };
+      request.application_properties![Constants.trackingId] = generate_uuid();
+      log.mgmt("[%s] Get session state request body: %O.", this._context.namespace.connectionId,
+        request.body);
+      log.mgmt("[%s] Acquiring lock to get the management req res link.",
+        this._context.namespace.connectionId);
+      await defaultLock.acquire(this.managementLock, () => { return this._init(); });
+      const result = await this._mgmtReqResLink!.sendRequest(request);
+      return result.body["session-state"]
+        ? this._context.namespace.dataTransformer.decode(result.body["session-state"])
+        : result.body["session-state"];
+    } catch (err) {
+      const error = translate(err);
+      log.error("An error occurred while sending the renew lock request to $management " +
+        "endpoint: %O", error);
+      throw error;
+    }
+  }
+
+  /**
+   * Lists the sessions on the ServiceBus Queue/Topic.
+   * @param lastUpdateTime Filter to include only sessions updated after a given time.
+   * @param skip The number of sessions to skip
+   * @param top Maximum numer of sessions.
+   * @returns Promise<ListSessionsResponse> A list of sessions.
+   */
+  async listMessageSessions(skip: number, top: number, lastUpdatedTime?: Date): Promise<ListSessionsResponse> {
```

Comment:
.net hasn't implemented this yet. they will be doing it soon. However Java has done this.

Reply:
Both the older .Net SDK and the current Java SDK have implemented this feature, but without the `skip` and the `top` arguments. Why not stay consistent with them, and add the 2 arguments as per feedback later in the cycle?

Reply:
Was this in the management client (ATOM based XML apis)? Since that is different than the current $management endpoint operation over AMQP.

Was able to find the similar AMQP implementation in java over [here](https://github.com/Azure/azure-service-bus-java/blob/dev/azure-servicebus/src/main/java/com/microsoft/azure/servicebus/primitives/MiscRequestResponseOperationHandler.java#L104).

The only thing that is different is the parameter lastSessionId. This is being set as a property on the request body if present. However, this property is not documented in the [docs](https://github.com/Azure/azure-service-bus-java/blob/dev/azure-servicebus/src/main/java/com/microsoft/azure/servicebus/primitives/MiscRequestResponseOperationHandler.java#L104).

The current implementation as-is works fine, however will have to talk to the service team to get a clear picture on this.

Reply:
Keeping this as-is right now. The current implementation works fine. We can definitely change it later.